driven/storage/adapter.go

```diff
@@ -342,6 +342,40 @@ func filterArgs(filter bson.M) *logutils.FieldArgs {
 	return &args
 }
 
+// FindScores finds scores based on filter
+func (a *Adapter) FindScores(filter bson.M, limit *int, offset *int) ([]model.Score, error) {
+	var scores []model.Score
+	err := a.db.scores.Find(a.context, filter, &scores, nil)
+	if err != nil {
+		return nil, errors.WrapErrorAction(logutils.ActionFind, model.TypeScore, filterArgs(filter), err)
+	}
+
+	return scores, nil
+}
+
+// UpdateScoreByFilter updates a score based on filter
+func (a *Adapter) UpdateScoreByFilter(filter bson.M, update bson.M) error {
+	_, err := a.db.scores.UpdateOne(a.context, filter, update, nil)
+	if err != nil {
+		return errors.WrapErrorAction(logutils.ActionUpdate, model.TypeScore, filterArgs(filter), err)
+	}
+
+	return nil
+}
+
+// BuildCoreAccountCriteriaByMastodonID builds MongoDB query criteria for finding Core BB accounts by mastodon_id
+// This returns a map that can be used with RetrieveCoreUserAccountByCriteria
+func BuildCoreAccountCriteriaByMastodonID(mastodonID string) map[string]interface{} {
+	return map[string]interface{}{
+		"identifiers": map[string]interface{}{
+			"$elemMatch": map[string]interface{}{
+				"code":       "mastodon_id",
+				"identifier": mastodonID,
+			},
+		},
+	}
+}
+
```

Comment:
Sorry for the confusion on this one. The issue here is that we are sending data that is strictly tied to mongo to the Core BB. Take a look at https://github.com/rokmetro/core-building-block-fork/blob/fc548a901e7b0b6d227cefc79b2343b675323690/driven/storage/adapter.go#L3452 to see what this map should look like.

I think you want something like
```
return map[string]interface{}{
	"identifiers": map[string]interface{}{
        "operation": "any",
		"value": map[string]interface{}{
			"code":       "mastodon_id",
			"identifier": mastodonID,
		},
	},
}
```

This new data structure will no longer be tied directly to mongo, so you can move it inside your `corebb.Adapter.RetrieveCoreUserAccountByCriteria` function because it is only used there.